doc/the-boot-partition-and-me.md

```diff
@@ -0,0 +1,82 @@
+## What is the boot partition used for and why would I need one?
+
+
+> #### Notes
+> - In this document only file systems supported by `storage-ng` are considered: btrfs, extX (ext2, ext3, ext4), xfs.
+> - With 'boot partition' a partition mounted at `/boot` (or containing the `/boot` directory directly) is meant.
+>   It is not to be confused with the BIOS GRUB partition, EFI System Partition, or PReP partition.
+
```

Comment:
I would also add a note specifying this document describes only the situation with BIOS (aka legacy) booting and MS-Dos partition tables. Just to avoid confusion.